barretenberg/cpp/src/barretenberg/ecc/groups/affine_element_impl.hpp

```diff
@@ -260,12 +264,11 @@ affine_element<Fq, Fr, T> affine_element<Fq, Fr, T>::hash_to_curve(const std::ve
         target_seed.push_back(0);
     }
     target_seed[seed_size] = attempt_count;
-    target_seed.back() = 0;
-    const auto hash_hi = sha256::sha256(target_seed);
-    target_seed.back() = 1;
-    const auto hash_lo = sha256::sha256(target_seed);
-    // custom serialize methods as common/serialize.hpp is not constexpr
-    // (next PR will make this method constexpr)
+    target_seed[seed_size + 1] = 0;
+    const auto hash_hi = blake3::blake3s_constexpr(&target_seed[0], target_seed.size());
```

Comment:
Note: we've changed the hash function presumably because we wanted this to be constexpr and because blake3 is faster

Reply:
I actually changed to blake3s because it's faster to implement *in-circuit*. i.e. if a circuit needs to evaluate hash-to-curve for whatever reason.